app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.taskyy.domain.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.ViewModel
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navigation
+import com.example.taskyy.ui.LoginScreen
+import com.example.taskyy.ui.RegisterScreen
+import com.example.taskyy.ui.viewmodels.LoginViewModel
+
+
+@Composable
+fun Nav() {
+
+    val navController = rememberNavController()
+    NavHost(navController = navController, startDestination = "login_screen" ) {
+        navigation(
+            startDestination = Screen.Login.route,
+            route = "login_screen"
+        ) {
+            composable(route = Screen.Login.route) {
+                val viewModel = hiltViewModel<LoginViewModel>()
+                val state = viewModel.state
+                LoginScreen(state, viewModel::onEvent, navController)
+            }
+            composable(route = Screen.Register.route) { entry ->
+                val loginViewModel = entry.loginViewModel<LoginViewModel>(navController = navController)
+                val state = loginViewModel.state
```

Comment:
I assume this is temporary since you've asked a question about this. In case not, you'd rather want to give the RegisterScreen its own ViewModel to keep the responsibilities separate. Login doesn't need to know anything about registration and registration doesn't need to know anything about login, yet with one shared VM you mix these together.

Shared ViewModels only make sense if you need to share state between multiple screens. Here each screen has its own state, so that's not needed